controllers/vrg_volrep.go

```diff
@@ -1773,10 +1780,10 @@ func (v *VRGInstance) addArchivedAnnotationForPVC(pvc *corev1.PersistentVolumeCl
 }
 
 // findProtectedPVC returns the &VRG.Status.ProtectedPVC[x] for the given pvcName
-func (v *VRGInstance) findProtectedPVC(pvcName string) *ramendrv1alpha1.ProtectedPVC {
+func (v *VRGInstance) findProtectedPVC(pvcNamespaceName, pvcName string) *ramendrv1alpha1.ProtectedPVC {
```

Comment:
When we upgrade to this version of the code and status, older VRGs that have been reconciled will not have the namespace field in protected PVCs. This function will hence always fail to find a protected PVC that is part of the status list.

Now, what I see is that the function `updateProtectedPVCs` would update the list and add 2 entries for the same PVC, one with the name and namespace and one pre-existing without the namespace.

There seems to be a gap in addressing the upgrade case here. Posting the concern as I maybe missing something here, and also to understand if this was tested?

Reply:
Good point.  The upgrade case is untested.  I'll plan to address that this week.   Logically the empty namespace is equivalent to the VRG's.

Reply:
Empty ProtectedPVC namespace values are now set right after VRG.Status is saved to address the upgrade case.

Reply:
A test case was added to recover from a backup without PVC namespace name in S3 path or VRG Status ProtectedPVCs.  It passes.